apps/src/tutorialExplorer/tutorialSet.jsx

```diff
@@ -24,8 +35,27 @@ export default class TutorialSet extends React.Component {
   tutorialClicked = item =>
     this.setState({showingDetail: true, chosenItem: item});
 
-  tutorialDetailClosed = () =>
-    this.setState({showingDetail: false, chosenItem: null});
+  tutorialDetailClosed = () => {
+    // Okay, so while the tutorialDetail window is open, it's possible that the focus has changed.
+    // The user could've hit left/right arrows while the detail was open and moved focus, or also could've
+    // changed focus into the dialog, so before we wipe out the chosenItem, we find the associated
+    // item in the tutorial list and focus it, and also make sure to scroll it into view because it's possible
+    // the user navigated through enough of them to be off screen.
+    const tutorialDiv = document.querySelectorAll(
+      `[data-tutorial-code="${this.state.chosenItem.code}"]`
+    )[0];
+
+    this.setState({showingDetail: false, chosenItem: null}, () => {
+      if (document.activeElement !== tutorialDiv) {
+        setTimeout(() => {
+          tutorialDiv.focus();
+          if (!isInViewport(tutorialDiv)) {
+            tutorialDiv.scrollIntoView({behavior: 'smooth', block: 'nearest'});
+          }
+        }, 0);
+      }
+    });
```

Comment:
There is a timing issue here, which seems to be related to how tightly `focus-trap-react` is grabbing focus and exactly when it's releasing it.

Best option I was able to find was to fall into the `setState` callback (deferring until state is properly fully updated) and _still_ needing a timeout to defer to the next pass of the event loop.